src/stream_python_file_by_line.c

```diff
@@ -98,14 +95,13 @@ _fb_load(void *fb)
     if (!FB(fb)->reached_eof && (FB(fb)->current_buffer_pos == FB(fb)->linelen)) {
         // Read a line from the file.
         //printf("_fb_load: calling readline\n");
-        PyObject *line = PyObject_Call(FB(fb)->readline, FB(fb)->empty_tuple, NULL);
+        PyObject *line = PyObject_CallFunctionObjArgs(FB(fb)->readline, NULL);
         //printf("_fb_load: back from readline\n");
-        FB(fb)->line = line;
+        Py_XSETREF(FB(fb)->line, line);
```

Comment:
Just for my own clarification: this is necessary because `FB(fb)->line` is a PyObject, right? Then the `Py_XSETREF` call essentially says: "set `FB(fb)->line` to `line`, then xdecref line, right? So the leak that this fixes is that line wasn't properly decrefed?

Sorry if the questions are inane, I have very little practice getting refcounting correct.

Also, re: why I was so confused when you were mentioning `Py_SETREF` before - it's because I'd never seen in in the C/API docs (apparently it's not officially public?). Searching online yields [object header file](https://github.com/python/cpython/blob/main/Include/cpython/object.h) as the "official" source of documentation :)

Reply:
> set FB(fb)->line to line

Yes, where `FB(fb)->line` is a strong reference (owns a reference count), or is NULL (the X).  It does not change the refcount for `line` (which I don't think you really meant), but decref's `FB(fb)->line`.

There is some subtleties to it if the decref might leave an object in a bad state, then you need to set the new object before the decref, although that usually doesn't matter, it is still a nice macro.